### PR TITLE
Riot Shield Updates

### PR DESCRIPTION
**Pull request by Super3222**
Refer to issue #1059 and #1055.
### Intent of your Pull Request
- Eshields now block disarms, like it should have been.
- Hulks during harm mode will bypass shields, ignoring them, and still doing damage to the blocker.
- So when hulks bypass shields they have a 50% chance to knock a shield out the blockers hand or the blocker will be weakened from the force of the hulks punch. So basically backfiring.
- Melee attacks that do more than 25 damage will break shields, shattering them and making them useless.

:cl:Super3222
rscadd: The Syndicate has noticed a faulty in their energy shield products which would make them disarmable. This was patched and they are now greytider resistant.
rscadd: Nanotrasen security has fallen victim towards their companies budget cuts, their riot shields can no longer block every melee attack. Be on the lookout for anything that might be too powerful and break em!
rscadd: Nanotrasen geneticists are SHOCKED after discovering that hulks are strong enough to instantly disarm shields and even throw them out of peoples hands! With the same attack, they can possibly make a blocker lose balance from their punch!
/:cl:

### CODE REVIEW

**Comment by ShadowDeath6** [comment]
> eshields are now immune to the graytide
> 
> :^^^^^^^^^^^^^^^^^^^^)

probably should change "em" to "them" in the last CL entry also

**Comment by FluffySurvivor** [comment]
> Melee attacks that do more than 25 damage will break shields, shattering them and making them useless.

Strongly disagree with this one. 

> 50% chance to knock

You should make it consistent, the disarm part should be guaranteed.

**Comment by ParagonPrime** [comment]
Eh. Mixed feelings on the 25-or-more-breaks-it thing.

On the one hand, eswords (lightsabers) should be able to deal with riot shields ez pz.

On the OTHER hand, it now makes it even harder or sec to stop dual eswords- they have to close to melee range as it is, but now they have to do that with even less protection.

**Comment by Super3222** [comment]
@ShadowDeath6:

> probably should change "em" to "them" in the last CL entry also

Sure.

@FluffySurvivor:

The riot shield buff got completely out of hand, because it was so unrealistic that it actually hurt to watch a security officer be able to block 4 juggernaughts surrounding him because he had one shield up in his hand.

There are only a few weapons that do more than 25 brute damage, but there are some which realistically wouldn't be able to block a riot shield. I'll see to making some arrrangements.

It's not just disarm though, it throws the shield out of their hands and sends it flying for close to 7 tiles.

@ParagonPrime:

Weapons with a force of over 25 will only break riot shields.
Energy Swords and Dual-Energy swords already CURRENTLY bypass shields due to having the flag NOSHIELD (which cancels out all blocking).

Subjected to traitors with dual-eswords, sec has a variety of options depending on what protective equipment the traitor has on.

To nukeops, they always had to go into close range.

**Comment by X-TheDark** [comment]
For all intents and purposes, this change only affects Cult (Juggernauts, who do 30 damage and Cult Blades, which have 30 Force) and Hulk (specific check), barring the blocking changes, since, as Super stated, energy swords already bypass shields (whether they should break them or not is another question, due to the bypass thing).

Here's a list of things that do more than 25 damage/force (may not be comprehensive):
- Juggernauts (30)
- Cult Blades (30)
- Claymore/Katana/Energy Katana (40)
- Esword (30)/Dual Esword (34) - bypasses shields entirely already
- Combat Mechas (30+)

Actually, I'm pretty sure that's it. Special mention goes to Spears that do 18 damage when wielded but bypass shields. Fireaxe does 24 damage when wielded, so riot shield still protects completely against that.

**Comment by ghost** [comment]
instead of sthattering the shield, make them bypass it, and shatter it after multiple consecutive hits

juggs still need to be blocked by riot shields though

**Comment by FluffySurvivor** [comment]
I still don't see why we should buff cult in this pr tho.

**Comment by Super3222** [comment]
> instead of sthattering the shield, make them bypass it, and shatter it after multiple consecutive hits

That was the original plan, and I have no idea where it derailed at. Shouldn't have been coding while I was exhausted last night.

> I still don't see why we should buff cult in this pr tho.

They were never meant to be nerfed in the first place. **Wraiths** (not harvesters, sorry about that) will still be blocked by riot shields though.

**Comment by ghost** [comment]
Why

Harvesters need to be the strongest construct they suck ass right now

**Comment by Super3222** [comment]
I meant Wraiths.

**Comment by EMT321** [comment]
Why are we buffing construct cult? As a person who enjoys it, there is already a counter: artificers. 

Why punish sec for having a rarely used counter? I can count 4 other instances where I've seen other people use shields vs cult, over the past two weeks.

Reactively nerfing is bad.

**Comment by ShadowDeath6** [comment]
Artificers currently have 40 or so health. They will guaranteed die from any lethal shotgun shell, fire of any kind, or a few lasers. They should not be imperative for the STRONGEST AVAILABLE CONSTRUCT to do damage whatsoever to one shitdick with a shield.

File: code/game/objects/items/weapons/shields.dm [hunk 1]
```diff
@@ -40,6 +51,19 @@
 		return ..()
 	else if(attack_type == PROJECTILE_ATTACK)
 		return ..()
+	else if(attack_type == MELEE_ATTACK && damage > block_limit)
+		shatter_reaction(owner)
```
Comment (oisin100) [inline code/game/objects/items/weapons/shields.dm]: Hmm, Perhaps give shield a certain amount of health. Any damage what exceeds the block limit will damage the shields health, once health reaches 0 shield should shatter.

**Comment by EMT321** [comment]
I agree with oison here. Making hulk+juggernauts an instant destruction of shields will confuse players and leave little room for getting away/counterplay.

**Comment by Super3222** [comment]
what adam el tablawy wrote:

> instead of sthattering the shield, make them bypass it, and shatter it after multiple consecutive hits

**Comment by Super3222** [comment]
@EMT321, Hulks don't instantly destroy shields.... come on now, second and third bullet point.

Attacks over 25 do... and lets take a second to consider what Juggernauts are.

Giant mass-shifting constructs that are suppose to hit like a truck (they do literally 30 brute damage, which is the amount energy swords hit for and energy shields pass shields) and be a tank, why should riot sheilds 100% of the time block every last hit that they throw. What would you call 9 juggernaughts trying to kill the one security officer cornered with a shield in his hand. Right, you'd call that unfair and unrealistic. They should be able to at least do damage to the shield.

I'll be taking every comment into consideration for the next update to this branch. I'll explain everything all over again, even. Also had the same thing in mind, @Oisin100.

**Comment by ShadowDeath6** [comment]
Eh, I'd rather the shield just fall on the ground than break entirely. They're extremely limited, and outright deleting them would be... Meh.

**Comment by Super3222** [comment]
> Eh, I'd rather the shield just fall on the ground than break entirely. They're extremely limited, and outright deleting them would be... Meh.

Cargo can always buy more. They are prematurely unlimited in amount, thus making it bad to completely counter every melee attack. Furthermore, R&D at combat 6 can replicate the hell out of them with enough silver.

**Comment by ShadowDeath6** [comment]
> relying on cargo
> 
> > > > > > > > > > relying on cargo's competence

**Comment by EMT321** [comment]
Destroying other people's items when they are used correctly is shitty. I can't think of another example, aside maybe from slings extinguishing lights, which is something few people enjoy. 

Make em force a disarm vs shields rather then outright delete them, and it makes a little more sense gameplay wise.

**Comment by EMT321** [comment]
And i have literally never seen someone other then myself order riot shields from csrgo.

**Comment by ghost** [comment]
cargo will never order riot shields don't kid yourself super

and R&D can mass produce teleshields, yes, but they can also mass produce stupidly retardedly absurdly lethal guns

no i think this is a fine PR

**Comment by Super3222** [comment]
Alright so:
Hulks will automatically throw the shields out of peoples hands, and with a fifty percent chance they will shove them to the ground as well.

Shields now have a health variable. Riot shields has 75 and the telescopic riot shield's have 95. Showing that the HoS's unique riot shield is unique until... R&D mass produces them.

What does this mean? Riot shields will still block any weapon/melee attack with a force of 25 or lower. However, anything stronger will damage it's health and bring it near the shattering point. Once it gets under 10, it will break.

**Comment by oisin100** [comment]
Dont supposed we can get some screenshots of the broken states?

**Comment by Super3222** [comment]
Here it is.
broke
broke2

**Comment by ShadowDeath6** [comment]
FIX CHANGELOG AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA

**Pull request merged**